example/voter/src/ibc.go

```diff
@@ -0,0 +1,150 @@
+package src
+
+import (
+	"bytes"
+	"errors"
+
+	"github.com/CosmWasm/cosmwasm-go/example/voter/src/state"
+	"github.com/CosmWasm/cosmwasm-go/example/voter/src/types"
+	"github.com/CosmWasm/cosmwasm-go/std"
+	stdTypes "github.com/CosmWasm/cosmwasm-go/std/types"
+)
+
+// handleIBCChannelOpen handles IBC channel open event.
+// Handles "Channel Open Init" / "Channel Open Try" and validates an IBC channel params.
+func handleIBCChannelOpen(msg stdTypes.IBCChannelOpenMsg) error {
+	switch {
+	case msg.OpenInit != nil:
+		// Do not check counterparty here, since it is not created yet (check on OpenTry later)
+		if err := types.ValidateIBCChannelParams(msg.OpenInit.Channel, false); err != nil {
+			return errors.New("openInit: " + err.Error())
+		}
+	case msg.OpenTry != nil:
+		if err := types.ValidateIBCChannelParams(msg.OpenTry.Channel, true); err != nil {
+			return errors.New("openTry: " + err.Error())
+		}
+		if err := types.ValidateIBCVersion(msg.OpenTry.CounterpartyVersion); err != nil {
+			return errors.New("openTry: counterparty: " + err.Error())
+		}
+	default:
+		return errors.New("unknown IBC channel open request")
+	}
+
+	return nil
+}
+
+// handleIBCChannelOpen handles IBC channel connect event.
+// Handles "Channel Open Ack" validating an IBC channel params.
+func handleIBCChannelConnect(msg stdTypes.IBCChannelConnectMsg) (*stdTypes.IBCBasicResponse, error) {
+	switch {
+	case msg.OpenAck != nil:
+		if err := types.ValidateIBCVersion(msg.OpenAck.CounterpartyVersion); err != nil {
+			return nil, errors.New("openAck: counterparty: " + err.Error())
+		}
+	case msg.OpenConfirm != nil:
+	default:
+		return nil, errors.New("unknown IBC channel connect request")
+	}
+
+	return &stdTypes.IBCBasicResponse{}, nil
+}
+
+// handleIBCMsgVote handles MsgIBC.Vote msg.
+// Failure error is ignored since we are using standard "0x01" and "0x00" acknowledgements.
+func handleIBCMsgVote(deps *std.Deps, env stdTypes.Env, req types.IBCVoteRequest) (retResp *stdTypes.IBCReceiveResponse, retErr error) {
+	defer func() {
+		if retErr != nil {
```

Comment:
can't this be handled in the IBC execute function ?

Reply:
I'm not sure what `IBCExecute` is.

Reply:
https://github.com/CosmWasm/cosmwasm-go/pull/152/files#diff-f52e1b13f5df70d42c30cace65821a2d041faf38c885ae150138c5e35ae39c31R423

sorry I meant the IBC packet received function

Reply:
`handleIBCMsgVote` func is a business logic for a specific packet since on IBC receive event different msgs could be handled differently. It is like we have a bunch of handlers for `MsgExecute`.